src/Microsoft.Health.Dicom.Core/Features/ExtendedQueryTag/ExtendedQueryTagErrorsService.cs

```diff
@@ -0,0 +1,50 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Dicom;
+using EnsureThat;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Extensions;
+using Microsoft.Health.Dicom.Core.Features.Common;
+using Microsoft.Health.Dicom.Core.Messages.ExtendedQueryTag;
+
+namespace Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag
+{
+    public class ExtendedQueryTagErrorsService : IExtendedQueryTagErrorsService
+    {
+        private readonly IExtendedQueryTagErrorStore _extendedQueryTagErrorStore;
+        private readonly IDicomTagParser _dicomTagParser;
+
+        public ExtendedQueryTagErrorsService(IExtendedQueryTagErrorStore extendedQueryTagStore, IDicomTagParser dicomTagParser)
+        {
+            _extendedQueryTagErrorStore = EnsureArg.IsNotNull(extendedQueryTagStore, nameof(extendedQueryTagStore));
+            _dicomTagParser = EnsureArg.IsNotNull(dicomTagParser, nameof(dicomTagParser));
+        }
+
+        public async Task<GetExtendedQueryTagErrorsResponse> GetExtendedQueryTagErrorsAsync(string tagPath, CancellationToken cancellationToken = default)
+        {
+            string numericalTagPath = _dicomTagParser.TryParse(tagPath, out DicomTag[] tags, supportMultiple: false)
```

Comment:
Not your code, but TryParse seems odd, Peng/Will, why does this API take TagPath and out an array of DicomTag?

Reply:
I think @pengchen0692 can probably speak to this a bit better. It doesn't appear we're supporting any path today that can return multiple tags per path

Reply:
yes, it was intended to support multiple level tag path.